tiny_dnn/parameter.h

```diff
@@ -0,0 +1,138 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/framework/tensor.h"
+#include "tiny_dnn/util/util.h"
+
+namespace tiny_dnn {
+
+enum class parameter_type : int8_t { weight = 0x1, bias = 0x2 };
+
+class Parameter : public std::enable_shared_from_this<Parameter> {
```

Comment:
I need a shared pointer as I could not take ownership of a shared pointer of weight if I had to set it externally. Doing so required `reset` and it created SIGABRT's whle adding new parameters.